src/coreclr/tools/Common/TypeSystem/Ecma/EcmaField.cs

```diff
@@ -322,20 +323,41 @@ public static int GetFieldRvaValue(this EcmaField field)
         /// <summary>
         /// Retrieves the data associated with an RVA mapped field from the PE module.
         /// </summary>
-        public static byte[] GetFieldRvaData(this EcmaField field)
+        public static bool TryGetFieldRvaData(this EcmaField field, out byte[] data)
         {
             Debug.Assert(field.HasRva);
             int addr = field.MetadataReader.GetFieldDefinition(field.Handle).GetRelativeVirtualAddress();
-            var memBlock = field.Module.PEReader.GetSectionData(addr).GetContent();
-
             int size = field.FieldType.GetElementSize().AsInt;
+
+            // If this data blob has fixups, we shouldn't return the bytes.
+            DirectoryEntry vtableFixups = field.Module.PEReader.PEHeaders.CorHeader.VtableFixupsDirectory;
+            if (vtableFixups.Size > 0)
+            {
+                TypeSystemContext context = field.Context;
+                BlobReader reader = field.Module.PEReader.GetSectionData(vtableFixups.RelativeVirtualAddress).GetReader(0, vtableFixups.Size);
+                while (reader.Offset < reader.Length)
+                {
+                    int fixupAddr = reader.ReadInt32();
+                    int numFixups = reader.ReadInt16();
+                    reader.ReadInt16(); // fixup type
+
+                    int fixupSize = numFixups * context.Target.PointerSize;
+                    if (addr >= fixupAddr + fixupSize || addr + size <= fixupAddr)
+                        continue;
+
+                    data = null;
+                    return false;
+                }
+            }
+
+            var memBlock = field.Module.PEReader.GetSectionData(addr).GetContent();
             if (size > memBlock.Length)
                 throw new BadImageFormatException();
 
-            byte[] result = new byte[size];
-            memBlock.CopyTo(0, result, 0, size);
+            data = new byte[size];
+            memBlock.CopyTo(0, data, 0, size);
```

Comment:
`GetContent()` allocates an `ImmutableArray<byte>`. We can avoid the extra allocation by getting the field data like this:

```csharp
            data = field.Module.PEReader.GetSectionData(addr).GetReader().ReadBytes(size);
```

We can avoid more copies by returning say a `ReadOnlyMemory<byte>` instead of a `byte[]`, but that would require more refactorings.